lib/cocoapods-core/source/metadata.rb

```diff
@@ -47,7 +47,7 @@ def path_fragment(pod_name, version = nil)
                        hashed.slice!(0, length)
                      end
                    end
-        prefixes.concat([pod_name, version]).compact.join(File::SEPARATOR)
+        prefixes.concat([pod_name, version]).compact
```

Comment:
wouldn't this affect the existing sharing functionality of the master specs repo?

Reply:
also looks like this function is missing test coverage

Reply:
It only had one use before this, and that's in `Pod::Source` where it actually gets splat into `Pathname.join` so it's joined according to filesystem rules anyway - resulting in the correct path. I've checked and there are multiple test that encounter this code path.

I agree about adding the test that directly tests this method.

Reply:
Added.